lib/index.ts

```diff
@@ -454,6 +465,45 @@ export class DockerProgress {
 		return hash;
 	}
 
+	/** Build docker image calling onProgress with extended progress info regularly */
+	async build(
+		image: string,
+		onProgress: ProgressCallback,
+		options?: BuildOptions,
+	): Promise<string> {
+		// Create tar for build
+		const tarStream = tar.pack();
+		tarStream.entry({ name: 'Dockerfile' }, `FROM ${image}`, (err) => {
+			if (err) {
+				throw err;
+			}
+			tarStream.finalize();
+		});
+
+		const ignoreErrorEvents = !!options?.ignoreProgressErrorEvents;
+		const reporter = await this.getProgressReporter();
+		onProgress = reporter.pullProgress(onProgress);
+		const stream = await this.docker.buildImage(
+			tarStream,
+			options as ImageBuildOptions,
+		);
+		const hash = await awaitRegistryStream(
+			stream,
+			onProgress,
+			ignoreErrorEvents,
+		);
+
+		const keepParentImage = !!options?.keepParentImage;
+		if (!keepParentImage) {
+			// This makes the operation non-atomic. Unfortunately, this means that
+			// if the process breaks at this point, the parent image will remain
+			// tagged on the OS. There is no atomic way to let the engine do this
+			await this.docker.getImage(image).remove();
```

Comment:
I don't understand what this comment says, could you please elaborate? This makes the operation non-atomic in which sense? Which operation -- removing the image or the whole build? Would it address your concern if you wrapped the operation in `try` and removed the image in a `finally` block?

Reply:
By operation I mean the whole build, i.e. build (letting the image handle the pull) + delete the parent image. Since it requires two steps, that makes it non-atomic, since power could cut in the middle of the two operations leaving you with the parent image on device. 

I'll probably remove the comment since is much more of a worry for the supervisor (where I'm thinking of using this), and I already have an idea for a cleanup step that would address this edge case.

Reply:
I suppose the `keepParentImage` is also a big flag suggesting this module can't assume anything about cleanup and it's just trying to be helpful.

Reply:
That's a good way to put it. We could add a comment saying cleanup step is "best effort", if the function terminates correctly it means the image was cleaned but it something interrupt the process there are no guarantees. 

> I suppose the `keepParentImage` option is also a flag suggesting this module can't assume anything about cleanup and it's just trying to be helpful.

Reply:
Removed the `keepParentImage` option as it was incompatible with passing a stream to the `build` function